docker-compose.yaml

```diff
@@ -78,41 +78,18 @@ services:
       - ./dnsmasq/dnsmasq.leases:/var/lib/misc/dnsmasq.leases
 
   user-data:
-    image: alpine/curl
+    image: alpine:3
     volumes:
-    - ./pxe:/pxe
+      - ./pxe:/pxe
+    environment:
+      - MACHINE_HOSTNAME=${MACHINE_HOSTNAME:-machine}
+      - MACHINE_PASSWORD=${MACHINE_PASSWORD:-'$2a$12$6EyKyPDuHQcHIF7Abawva.pCLQhGs0r/3NybeiMVgaKTDPbm.a7b6'}
+      - MACHINE_USERNAME=${MACHINE_USERNAME:-contestadmin}
+      - ACNG_PORT=${ACNG_PORT:-3142}
     command:
       - /bin/sh
       - -c
       - |
-        set -o pipefail && 
-        cat<<"EOT">/pxe/user-data
-        #cloud-config
-        autoinstall:
-          version: 1
-          drivers:
-            install: true
-          proxy: http://_gateway:3142
-          kernel:
-            flavor: hwe
-          power_state:
-            mode: reboot
-            message: Install done, bye bye
-          identity:
-            hostname: ${MACHINE_HOSTNAME:-machine}
-            password: ${MACHINE_PASSWORD:-'$2a$12$6EyKyPDuHQcHIF7Abawva.pCLQhGs0r/3NybeiMVgaKTDPbm.a7b6'}
-            realname: ${MACHINE_USERNAME:-contestadmin}
-            username: ${MACHINE_USERNAME:-contestadmin}
-          storage:
-            layout:
-              name: lvm
-            grub:
-              reorder_uefi: True
-              update_nvram: True
-              reorder_uefi_force_fallback: True
-          ssh:
-            allow-pw: false
-            install-server: true
-            authorized-keys: 
-        EOT
-        echo "${GITHUB_USERNAMES:-tuupke}" | tr "," "\n" | (xargs -I{} curl -Sf https://github.com/{}.keys | sed -e 's/^/    - /') >>/pxe/user-data 
+        apk -U add gettext curl
+        envsubst < pxe/user-data.template | tee pxe/user-data
+        echo "${GITHUB_USERNAMES:-tuupke}" | tr "," "\n" | (xargs -I{} curl -Sf https://github.com/{}.keys | sed -e 's/^/    - /') >> /pxe/user-data
```

Comment:
Maybe we can get rid of the 'tuupke' part by adding an environment variable where you can add a single public key which defaults to the contents of `~/.ssh/id_rsa.pub` (any other key locations) or a hardcoded public key for which we hold the private key to hand out if need be. (which is again the only reason why my gh username is in the 'default' list.